### DISCUSSION TITLE
I want to extract the result values of Hammerdbcli executed through a Python script as a dictionary variable.

### DISCUSSION THREAD

**Discussion by Rayjun0525** [TPC-Council/HammerDB#575] [Q&A] [answered]
I want to assign the result values of hammerdbcli executed using a Python script to a dictionary variable. I am planning overall automation and am thinking about assigning the results to a variable for reprocessing after running hammerdb through python. I am curious about how to run hammerdb within a python script. I have tried using ‘hammerdbcli py’ interactively, but I could not find an example of how to automate it within a script. I am thinking of using subprocess to operate based on the shell, but it doesn’t seem like a good idea, so I am asking. Also, I want to bring the result value into a dictionary variable, but I can’t think of a good idea. Can you guide me?

**Comment by sm-shaw** [comment, marked as answer]
There are examples included within HammerDB of how to run within a script with Python that can be found under this directory:
https://github.com/TPC-Council/HammerDB/tree/master/scripts/python
Following the examples here you can see how you can move easily between Python and HammerDB (Note this article explains why we don't use Python for the actual HammerDB workload https://www.hammerdb.com/blog/uncategorized/why-tcl-is-700-faster-than-python-for-database-benchmarking/)
The scripts show examples of how you can redirect output to files and then read them into Python or for specific examples you can use the tclpy.eval command to call a command directly in HammerDB and set the variable in Python.  
```
hammerdb>>>x = tclpy.eval('set x [expr 10 + 5]')
hammerdb>>>print(x)
15
hammerdb>>>
```
